crates/prices/src/core/estimate.rs

```diff
@@ -0,0 +1,84 @@
+use {
+    super::{estimator, eth, swap, Estimator, Swap},
+    futures::future::join_all,
+};
+
+/// The estimated outcome of a [`Swap`]. Currently, this is simply the median
+/// value of the [`estimator::Estimate`]s returned by the [`Estimator`]s.
+#[derive(Debug, Clone, Copy)]
+pub struct Estimate {
+    /// The [`Swap`] that was estimated.
+    pub swap: Swap,
+    /// The amount of [`ToToken`] that the user receives.
+    pub to: swap::ToAmount,
+    /// The amount of [`ToToken`] paid in fees.
+    pub fee: swap::ToAmount,
+}
+
+/// Estimate the outcome of a [`Swap`].
+pub async fn estimate(
+    swap: Swap,
+    deadline: Deadline,
+    estimators: &[Box<dyn Estimator>],
+) -> Result<Estimate, Error> {
+    // Fetch the estimate for the swap.
+    let estimate = median_estimate(swap, deadline, estimators).await?;
+
+    // Convert the fee from ETH to [`ToToken`].
+    let fee = median_estimate(
+        Swap {
+            from: eth::ETH_TOKEN.into(),
+            to: swap.to,
+            amount: estimate.gas.0.into(),
+        },
+        deadline,
+        estimators,
+    )
+    .await?;
```

Comment:
These need to be joined otherwise an `estimate` call with `deadline: 3s` could take 6s in total.
Also it's nice to reduce the latency for when users request a price.

Reply:
This got joined in one of the following PRs.